### ISSUE TITLE
Can we get hooked up to the node.js CI infrastructure?

### ISSUE THREAD

**Issue by kkoopa** [nodejs/nan#508]
As @bnoordhuis [mentioned](https://github.com/nodejs/nan/issues/507#issuecomment-153419397), it might be possible to get us hooked up to the node.js CI infrastructure.

In general, this could be very nice.

From a personal point of view, my main concerns are with ease of use and speed.
- Will it be at least as fast as Travis, given queues and whatnot?
- Is it possible to hook the CI infrastructure into the GitHub API?

Are CI builds done automatically or manually in node.js?
If they are started manually, the second point above is less important. We could then rely on the current setup for everyday use and do occasional manual checks on a big-endian machine before publishing new versions or merging PRs which might run into endianness problems.

I guess a manual build setup would be easier for the node.js infrastructure team to pull off and would produce less overhead and noise.

/cc @rvagg

**Comment by bnoordhuis** [comment]
> Will it be at least as fast as Travis, given queues and whatnot?

Yes in general.  Some of the arm buildbots are pretty slow though.

**Comment by kkoopa** [comment]
> Some of the arm buildbots are pretty slow though.

If they run in parallel, I don't think it should be too much of a problem. If it is less than five minutes, it should probably be fine. If we end up only doing occasional one-off builds on a big-endian machine, the speed should be of no concern. Also, building NAN's test suite should be a far cry from building V8 + node.js + libuv.

**Comment by tniessen** [comment]
See also https://github.com/nodejs/nan/issues/958.
